diff --git a/Synthetic_benchmarks.py b/Synthetic_benchmarks.py
index e3c9848..4d2e9fb 100644
--- a/Synthetic_benchmarks.py
+++ b/Synthetic_benchmarks.py
@@ -131,6 +131,9 @@ class RandomLabelMemorizationDataset(InMemoryDataset):
             
             data = from_networkx(G_p)
             data.x = torch.ones((data.num_nodes, 1)) # Feature all 1
+            # graph_id = torch.zeros(self.num_graphs)
+            # graph_id[i] = 1.0
+            # data.x = graph_id.repeat(data.num_nodes, 1) # Unique identifier for each graph as node features
             data.y = torch.tensor([targets[i]], dtype=torch.long) # Label 
             
             # Metadata saved in the graph data object
@@ -309,9 +312,9 @@ class ParentClassificationDataset(InMemoryDataset):
 
 if __name__ == "__main__":
    
-    # ### MEMORIZATION TASK 
-    # for k in [0, 1, 2, 3, 4, 5]:
-    #     dataset = RandomLabelMemorizationDataset(root='./data/memorization', regime='all', K=k)
+    ### MEMORIZATION TASK 
+    for k in [0, 2, 5]:
+        dataset = RandomLabelMemorizationDataset(root='./data/memorization', regime='all', K=k, num_graphs=20)
 
     # for r in [0.1, 0.2, 0.4, 0.6, 0.8, 1.0]:
     #     dataset = RandomLabelMemorizationDataset(root='./data/memorization', regime='fraction', K=3, rho=r)
diff --git a/run_perturb.py b/run_perturb.py
index 69bed32..d048c3c 100644
--- a/run_perturb.py
+++ b/run_perturb.py
@@ -14,29 +14,31 @@ from torch_geometric.transforms import Compose
 
 from tqdm import tqdm
 from src.utils import OneHotNodeLabel, CalculateWLColors
-from src.net import Net, GCN
+from src.net import Net, GCN, GIN
 from collections import defaultdict
 import networkx as nx
 import os
 from Synthetic_benchmarks import ParentClassificationDataset, RandomLabelMemorizationDataset
 
 batch_size = 16
-num_layers = [1, 2, 3, 4, 5, 6]
+num_layers = [1,2,3,4,5,6]
 lr = 0.0001
-epochs = 500
 lipschits_constant_loss = 1.0
 delta_prob = 0.95
 #seeds = [1, 2, 3, 4, 5] 
 num_reps = 5
-hd = 16
-early_stopping = 20 
+##from https://openreview.net/pdf?id=C6vpifaZvU 
+hd = 64
+epochs = 3000  
+early_stopping = 200
+
 
 
 def init_dataset(dataset_name, transforms, path, k, rho=1.0):
     k = int(k)
     if dataset_name == 'memo_all_pert':
         # ### MEMORIZATION TASK 
-        dataset = RandomLabelMemorizationDataset(root=os.path.join(path, 'memorization'), num_graphs=1000, pre_transform=transforms, regime='all', K=k)
+        dataset = RandomLabelMemorizationDataset(root=os.path.join(path, 'memorization'), num_graphs=100, pre_transform=transforms, regime='all', K=k)
     elif dataset_name == "memo_partial_pert":
         dataset = RandomLabelMemorizationDataset(root=os.path.join(path, 'memorization'), num_graphs=1000, pre_transform=transforms, regime='fraction', K=k, rho=rho)
 
@@ -94,7 +96,7 @@ def main(args):
             "rho": args.rho,
             "seed": args.seed
         }
-        run_name = f"GCN_{dataset_name}_k{args.k}_rho{args.rho}_L{l}"
+        run_name = f"GIN_{dataset_name}_k{args.k}_rho{args.rho}_L{l}"
         with wandb.init(name=run_name, project="wl_perturb", entity="wl_meet_rad", config=config) as run: #, mode="disabled"
             #dataset.shuffle()
             p_dict = defaultdict(int)
@@ -122,7 +124,7 @@ def main(args):
             test_loader = DataLoader(test_dataset, batch_size)
 
             # model = Net(dataset.num_features, hd, 1, l).to(device) # Binary
-            model = GCN(dataset.num_features, hd, 1, l).to(device) # Binary
+            model = GIN(dataset.num_features, hd, 1, l).to(device) # Binary
             print("Model params:", sum(p.numel() for p in model.parameters() if p.requires_grad))
 
             optimizer = torch.optim.Adam(model.parameters(), lr=lr)
diff --git a/wandb/latest-run b/wandb/latest-run
index 8955b7f..37932ee 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20260428_174304-apuli11j
\ No newline at end of file
+run-20260429_133637-4kik0f7v
\ No newline at end of file
